core/lib/eth_client/src/types.rs

```diff
@@ -176,4 +283,198 @@ mod tests {
+        ).unwrap();
+
+        let signer = PrivateKeySigner::new(private_key);
+
+        // A blob we want to test is the 12 bytes `'A'`. It gets
+        // expanded to the constant size of the blob since blobs in
+        // EIP4844 transactions are of constant size.
+        let mut blob = vec![0u8; 131072];
+
+        // set the blob first 12 bytes to `'A'`.
+        for item in blob.iter_mut().take(12) {
+            *item = b'A';
+        }
+
+        let raw_transaction = TransactionParameters {
+            chain_id: 9,
+            nonce: 0.into(),
+            max_priority_fee_per_gas: U256::from(1),
+            gas_price: Some(U256::from(4)),
+            to: Some(H160::from_str("0x0000000000000000000000000000000000000001").unwrap()),
+            gas: 0x3.into(),
+            value: Default::default(),
+            data: Default::default(),
+            transaction_type: Some(U64::from(EIP_4844_TX_TYPE)),
+            access_list: None,
+            max_fee_per_gas: U256::from(2),
+            max_fee_per_blob_gas: Some(0x4.into()),
+            blob_versioned_hashes: Some(vec![H256::from_slice(
+                hex::decode("01a034bbe3f441bdce53ea4bcf4aa3c7561bc03d4559fdb88e1d33a868e5c56e")
+                    .unwrap()
+                    .as_ref(),
+            )]),
+        };
+        let raw_tx = signer
+            .sign_transaction(raw_transaction.clone())
+            .await
+            .unwrap();
+
+        let hash = web3::signing::keccak256(&raw_tx).into();
+        // Transaction generated with https://github.com/inphi/blob-utils with
+        // the above parameters.
+        let expected_str = include_str!("testdata/4844_tx_1.txt");
+        let expected_str = expected_str.trim();
+
+        let signed_call_result = SignedCallResult::new(
+            RawTransactionBytes(raw_tx),
+            U256::from(2),
+            U256::from(1),
+            0.into(),
+            hash,
+        );
+
+        let raw_tx_str = hex::encode(
+            signed_call_result.raw_tx(
+                Some(EthTxBlobSidecar::EthTxBlobSidecarV1(EthTxBlobSidecarV1 {
+                    blobs: vec![SidecarBlobV1 {
+                        blob,
+                        commitment,
+                        proof,
+                    }],
+                }))
+                .as_ref(),
+            ),
+        );
+
+        let expected_str = expected_str.to_owned();
+        pretty_assertions::assert_eq!(raw_tx_str, expected_str);
+    }
+
+    #[tokio::test]
+    // Tests the encoding of the `EIP_4844_TX_TYPE` transaction to
+    // network format defined by the `EIP`. That is, a signed transaction
+    // itself and the sidecar containing the blobs.
+    async fn test_generating_signed_raw_transaction_with_4844_sidecar_two_blobs() {
+        let private_key =
+            H256::from_str("27593fea79697e947890ecbecce7901b0008345e5d7259710d0dd5e500d040be")
```

Comment:
i hope it's newly generated key.

Reply:
https://github.com/matter-labs/zksync-era/blob/636fcfd308cdde92c6407b886d7b329938623e9d/core/lib/env_config/src/eth_sender.rs#L79